QuietFactory.php

```diff
@@ -49,11 +48,21 @@ public function createItem($name, array $options = array())
         try {
             return $this->innerFactory->createItem($name, $options);
         } catch (RouteNotFoundException $e) {
-            $this->logger->error(sprintf('%s : %s', $name, $e->getMessage()));
+            if (null !== $this->logger) {
+                $this->logger->error(
+                    sprintf('An exception was thrown while creating a menu item called "%s"', $name),
+                    array('exception' => $e)
+                );
+            }
 
             if (!$this->allowEmptyItems) {
                 return null;
             }
+
+            // remove route to create an item without any URL
+            unset($options['route']);
```

Comment:
could it be something else like the content of the menu item that is RouteReferrersReadInterface but does not have any route associated? or some random content that does not even have the interface? we should also remove other things that our factory might be looking at.